metaquot/metaquot.ml

```diff
@@ -93,8 +93,12 @@ let quote_of_path (path : Path.t) : Ppxlib.expression =
   let name =
     match Untypeast.lident_of_path path with
     | Lident name | Ldot (Lident "Asttypes", name) -> name
-    | Ldot (Lident "Location", "t") -> "location"
-    | Ldot (Lident "Longident", "t") -> "longident"
+    | Ldot (Lident "Location", "t") (* ppxlib < 0.23.0 *)
+    | Ldot (Ldot (Lident "Astlib__", "Location"), "t") -> (* ppxlib >= 0.23.0 *)
```

Comment:
By any chance, is it possible to use the public path to `Location`, so `Ppxlib_astlib.Location.t` , instead of the private one, i.e. `Astlib__.Location.t`, here? And in case that's possible, same for `Longident` below.